library/server.c

```diff
@@ -183,10 +178,8 @@ struct dnet_node *dnet_server_node_create(struct dnet_config_data *cfg_data)
 	pthread_sigmask(SIG_SETMASK, &previous_sigset, NULL);
 	return n;
 
-	dnet_set_need_exit(n);
```

Comment:
Why was it removed?

Reply:
Got it

Reply:
But it is set before everything to force threads to stop.

Reply:
> But it is set before everything to force threads to stop.

If so, we must to do `dnet_set_need_exit` after each error label. But this logic isn't done in source version.

Also, result behavior can be simply tested, by breaking process manually and try all possible scopes of init-deinit.

Reply:
If it is possible, could it be covered by some test?

Reply:
> But it is set before everything to force threads to stop.

```
dnet_backends_cleanup_all
dnet_route_list_destroy
dnet_local_addr_cleanup
```
These methods don't use `need_exit`. And `dnet_set_need_exit` stops threads eventually and doesn't join them. I think, that does not matter, where this line will be.